bin/faucet/frontend/app.js

```diff
@@ -192,6 +200,54 @@ export class MidenFaucetApp {
         }
     }
 
+    pollNote(noteId) {
+        return new Promise((resolve, reject) => {
+            // Poll every 500ms for the first 10 seconds, then every 1s for the next 30 seconds, then every 5s.
+            let currentInterval = 500;
+            let pollInterval;
+            let elapsedTime = 0;
+            // Timeout after 5 minutes
+            const timeout = 300000;
+            let timeoutId;
+
+            const poll = async () => {
+                try {
+                    const note = await this.rpcClient.getNotesById([NoteId.fromHex(noteId)]);
+                    if (note && note.length > 0) {
+                        clearInterval(pollInterval);
+                        clearTimeout(timeoutId);
+                        resolve();
+                        return;
+                    }
+
+                    elapsedTime += currentInterval;
+
+                    if (elapsedTime <= 10000) {
+                        currentInterval = 500;
+                    } else if (elapsedTime <= 40000) {
+                        currentInterval = 1000;
+                    } else {
+                        currentInterval = 5000;
+                    }
```

Comment:
Not sure how this is handled in JS, but it may be more legible to have a `SECOND` and `MINUTE` declared as constants:
```js
const SECOND = 1000;
const MINUTE = 60 * SECOND;
```

So we can have this:
```suggestion
                    if (elapsedTime <= 10 * SECOND) {
                        currentInterval = 0.5 * SECOND;
                    } else if (elapsedTime <= 40 * SECOND) {
                        currentInterval = 1 * SECOND;
                    } else {
                        currentInterval = 5 * SECOND;
                    }
```

(Also applies to the rest of the code)